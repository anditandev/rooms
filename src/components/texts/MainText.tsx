import React from 'react';
import { StyleSheet, Text, TextStyle} from 'react-native';

interface Props {
    style?: TextStyle | TextStyle[];
    text: string;
};

const MainText = (props: Props) => {
	return (
        <Text style={[styles.textStyle, props.style]}>
            {props.text}
        </Text>
	);
};

const styles = StyleSheet.create({
	textStyle: {
        fontSize: 14,
	},
});

export default MainText;
