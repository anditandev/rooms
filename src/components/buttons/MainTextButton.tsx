import React from 'react';
import MainText from '@components/texts/MainText';
import { Pressable, TextStyle, ViewStyle} from 'react-native';

interface Props {
    accessibilityLabel?: string;
    viewStyle?: ViewStyle | ViewStyle[];
    textStyle?: TextStyle | TextStyle[];
    pressHandler?: () => void;
    text: string;
};

const MainTextButton = (props: Props) => {

    const pressHandler = () => {
        if (props.pressHandler)
            props.pressHandler();
    };

	return (
        <Pressable 
            accessibilityLabel={props.accessibilityLabel} 
            onPress={pressHandler} 
            style={[props.viewStyle]}
        >
            <MainText style={props.textStyle} text={props.text} />
        </Pressable>
	);
};

export default MainTextButton;
