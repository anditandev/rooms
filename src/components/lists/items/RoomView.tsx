import React from 'react';
import { StyleSheet, View } from 'react-native';
import Animated from 'react-native-reanimated';
import MainText from '@components/texts/MainText';

interface Props {
    name: string;
    level: string;
    capacity: string;
    isAvailable: boolean;
};

const RoomView = (props: Props) => {

    const isAvailable = props.isAvailable;
    const availabilityTextColor = isAvailable ? '#52BA50' : '#939393';
    const availabilityText = isAvailable ? 'Available' : 'Not Available';

    return (
        <Animated.View style={[styles.container]}>
            <View>
                <MainText style={styles.nameTextStyle} text={props.name} />
                <MainText style={styles.levelTextStyle} text={`Level ${props.level}`} />
            </View>
            <View>
                <MainText 
                    style={[
                        styles.availabilityTextStyle,
                        {
                            color: availabilityTextColor,
                        },
                    ]} 
                    text={availabilityText} 
                />
                <MainText style={styles.capacityTextStyle} text={`${props.capacity} Pax`} />
            </View>
        </Animated.View>
    );
};

const styles = StyleSheet.create({
    container: {
        borderRadius: 8,
        backgroundColor: '#F7F7F7',
        marginTop: 8,
        flexDirection: 'row',
        paddingHorizontal: 24,
        paddingVertical: 16,
        justifyContent: 'space-between',
    },
    nameTextStyle: {
        fontSize: 16,
        lineHeight: 16,
        fontWeight: 'bold',
        color: '#4F4F4F',
    },
    levelTextStyle: {
        fontSize: 14,
        lineHeight: 16,
        color: '#4F4F4F',
        marginTop: 8,
    },
    availabilityTextStyle: {
        fontSize: 14,
        lineHeight: 16,
        textAlign: 'right',
        fontStyle: 'italic',
    },
    capacityTextStyle: {
        fontSize: 14,
        lineHeight: 16,
        color: '#4F4F4F',
        textAlign: 'right',
        marginTop: 8,
    },
});

const equalValues = (prevValue: Props, nextValue: Props) => {
    return prevValue.name === nextValue.name
    && prevValue.isAvailable === nextValue.isAvailable
    && prevValue.capacity === nextValue.capacity
    && prevValue.level === nextValue.level
};

export default React.memo(RoomView, equalValues);
