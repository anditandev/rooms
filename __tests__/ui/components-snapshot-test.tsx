import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MainTextButton from '@components/buttons/MainTextButton';
import MainText from '@components/texts/MainText';
import { Pressable, Text } from 'react-native';
import RoomView from '@components/lists/items/RoomView';

describe('Main Text component test', () => {

    const text = 'Text 1';
    const style = {
        fontSize: 16,
    };

    const textComponent = renderer
    .create(
        <MainText style={style} text={text} />
    )
    .toJSON();

    test('expect component to match snapshot', () => {
        expect(textComponent).toMatchSnapshot();
    });
});

describe('Main Button Text component test', () => {

    const text = 'Button 1';
    const accessibilityLabel = 'Press handler for button 1';
    const textStyle = {
        fontSize: 16,
        color: '#FFF',
    };

    const viewStyle = {
        marginTop: 12,
        backgroundColor: '#4E77E0',
        borderRadius: 25,
    };

    const fn = jest.fn();
    const buttonComponent = renderer
    .create(
        <MainTextButton
            pressHandler={fn}
            accessibilityLabel={accessibilityLabel}
            viewStyle={viewStyle}
            textStyle={textStyle}
            text={text}
        />
    )
    .toJSON();

    test('expect component to match snapshot', () => {
        expect(buttonComponent).toMatchSnapshot();
    });
});

describe('Room view component test', () => {

    const name = 'Test 1';
    const level = '3';
    const capacity = '10';
    const availability = false;

    const roomViewComponent = renderer
    .create(
        <RoomView
            name={name}
            level={level}
            capacity={capacity}
            isAvailable={availability}
        />
    )
    .toJSON();

    test('expect component to match snapshot', () => {
        expect(roomViewComponent).toMatchSnapshot();
    });
});

