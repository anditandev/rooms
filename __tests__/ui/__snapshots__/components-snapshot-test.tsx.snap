// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Button Text component test expect component to match snapshot 1`] = `
<View
  accessibilityLabel="Press handler for button 1"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#4E77E0",
        "borderRadius": 25,
        "marginTop": 12,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
        },
        Object {
          "color": "#FFF",
          "fontSize": 16,
        },
      ]
    }
  >
    Button 1
  </Text>
</View>
`;

exports[`Main Text component test expect component to match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 14,
      },
      Object {
        "fontSize": 16,
      },
    ]
  }
>
  Text 1
</Text>
`;

exports[`Room view component test expect component to match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F7F7F7",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 8,
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#4F4F4F",
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 16,
          },
        ]
      }
    >
      Test 1
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#4F4F4F",
            "fontSize": 14,
            "lineHeight": 16,
            "marginTop": 8,
          },
        ]
      }
    >
      Level 3
    </Text>
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Array [
            Object {
              "fontSize": 14,
              "fontStyle": "italic",
              "lineHeight": 16,
              "textAlign": "right",
            },
            Object {
              "color": "#939393",
            },
          ],
        ]
      }
    >
      Not Available
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#4F4F4F",
            "fontSize": 14,
            "lineHeight": 16,
            "marginTop": 8,
            "textAlign": "right",
          },
        ]
      }
    >
      10 Pax
    </Text>
  </View>
</View>
`;
